### ISSUE TITLE
Query freshness lag spikes after running rebalance w/ includeConsuming=false

### ISSUE THREAD

**Issue by dang-stripe** [apache/pinot#9784]
We're currently doing this for realtime table rebalances to minimize data ingestion lag:

1. Rebalance table w/ includeConsuming=false - instance partition map is updated w/ new assignments, but no new consuming segments are created
2. forceCommit - ensure all consuming segments are sealed and new consuming segments created w/ updated assignment
3. Rebalance table again w/ includeConsuming=false - ensure lingering consuming segments are moved to correct hosts

We've noticed large spikes (within bounds of the segment flush time threshold) in the broker's freshness_lag_ms metric for realtime tables after step 1 that don't go away until step 2 is complete. We suspect this might either be related to rebalance w/ includeConsuming=false or the metric may be reporting incorrectly.

cc @Jackie-Jiang

**Comment by dang-stripe** [comment]
Alternatively, we've tried this - this seems to avoid the issue above while having a slightly longer ingestion delay.

1. Force commit the table
2. Rebalance w/ includeConsuming=true

**Comment by Jackie-Jiang** [comment]
Checked the code and confirmed the behavior: when the committed segments are moved to a different server, when that server is queried, we can only use the max end time of the committed segments as the freshness time, which leads to this freshness lag spike. After the consuming segment is moved, the freshness lag will get back to normal.

This issue also applies to the tables that have committed segments relocated to a different server. We should find out a way to handle it properly.

Tl;dr: the spike is from freshness mis-calculation

**Comment by dang-stripe** [comment]
Closing this as a duplicate of https://github.com/apache/pinot/issues/11448
